app/controllers/uploads_controller.rb

```diff
@@ -20,10 +20,14 @@ def show
   def destroy
     @upload = Upload.find(params[:id])
     @upload.destroy
-    if current_user.admin?
+    if current_user.admin? || current_user.id == @upload.folder.owner.id
```

Comment:
Rather than calling `@upload.folder.owner.id`, can we create a method in the model?